weather-station-oracle/smart-contract/contracts/WeatherStationOracle.sol

```diff
@@ -0,0 +1,77 @@
+pragma solidity ^0.5.8;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "iexec-doracle-base/contracts/IexecDoracle.sol";
+
+contract WeatherStationOracle is Ownable, IexecDoracle
+{
+	struct WeatherReport
+	{
+		bytes32 oracleCallID;
+		string 	description;
+		bool	historicalData;
+	}
+
+	struct Location
+	{
+		string  location;
+		mapping(uint256 => WeatherReport) reports;
```

Comment:
it would be nice to have reports as an array of WeatherReport (each WeatherReport containing the associated time).
That way we can traverse the history / get the latest value easily.

Another solution is to only store the latest report and rely on offchain services (such as TheGraph) to index past events

Reply:
I thought it could be nice to use an array as well. But as we allow users to pass a specific timestamp, the reports being fetched will not necessarily be in a correct time order, that's why I used a mapping instead. Don't you think it is better in this case ?